pkg/http/server.go

```diff
@@ -127,23 +129,102 @@ func Run(cfg config.MutualPeersConfig) {
 	log.Info("Server Exited Properly")
 }
 
-func GenerateHashMetrics(cfg config.MutualPeersConfig, err error) bool {
-	// Get the genesisHash
-	// check if the config has the consensusNode field defined
+// BackgroundGenerateHashMetric checks if the consensusNode field is defined in the config to generate the metric from the Genesis Hash data.
+func BackgroundGenerateHashMetric(cfg config.MutualPeersConfig) {
+	log.Info("BackgroundGenerateHashMetric...")
+
 	if cfg.MutualPeers[0].ConsensusNode != "" {
-		blockHash, earliestBlockTime := nodes.GenesisHash(cfg)
-		err = metrics.WithMetricsBlockHeight(
-			blockHash,
-			earliestBlockTime,
-			cfg.MutualPeers[0].ConsensusNode,
-			os.Getenv("POD_NAMESPACE"),
-		)
-		if err != nil {
-			log.Errorf("Error registering metric block_height_1: %v", err)
-			return true
+		log.Info("Initializing goroutine to generate the metric: hash ")
+
+		// Create an errgroup with a context
+		eg, ctx := errgroup.WithContext(context.Background())
+
+		// Run the WatchHashMetric function in a separate goroutine
+		eg.Go(func() error {
+			log.Info("Consensus node defined to get the first block")
+			return WatchHashMetric(cfg, ctx)
+		})
+
+		// Wait for all goroutines to finish
+		if err := eg.Wait(); err != nil {
+			log.Error("Error in BackgroundGenerateHashMetric: ", err)
+			// Handle the error as needed
 		}
 	}
-	return false
+}
+
+// WatchHashMetric watches for changes to generate hash metrics in the specified interval.
+func WatchHashMetric(cfg config.MutualPeersConfig, ctx context.Context) error {
+	// Create a new context derived from the input context with a timeout
+	ctx, cancel := context.WithTimeout(ctx, hashMetricGenTimeout)
+	defer cancel()
+
+	// Create an errgroup with the context
+	eg, ctx := errgroup.WithContext(ctx)
+
+	// Run the WatchHashMetric function in a separate goroutine
+	eg.Go(func() error {
+		// Continue generating metrics with retries
+		for {
+			select {
+			case <-ctx.Done():
+				// Context canceled, stop the process
+				log.Info("Context canceled, stopping WatchHashMetric.")
+				return ctx.Err()
+			default:
+				err := GenerateHashMetrics(cfg)
+				// Check if err is nil, if so, Torch was able to generate the metric.
+				if err == nil {
+					log.Info("Metric generated for the first block, let's stop the process successfully...")
+					// The metric was successfully generated, stop the retries.
+					return nil
+				}
+
+				// Log the error
+				log.Error("Error generating hash metrics: ", err)
+
+				// Wait for the retry interval before the next execution using a timer
+				timer := time.NewTimer(retryInterval)
+				select {
+				case <-ctx.Done():
+					// Context canceled, stop the process
+					log.Info("Context canceled, stopping WatchHashMetric.")
+					timer.Stop()
+					return ctx.Err()
+				case <-timer.C:
+					// Continue to the next iteration
+				}
```

Comment:
[nit]: can be a separate private func to keep it neat 
Can be parked as an issue to move on